components/dashboard/sections/transactions-imports/TransactionsImport.tsx

```diff
@@ -162,15 +169,36 @@ export const TransactionsImport = ({ accountSlug, importId }) => {
   useQuery(transactionsImportLasSyncAtPollQuery, {
     context: API_V2_CONTEXT,
     variables: { importId },
-    pollInterval: !importData?.lastSyncAt ? 2_000 : 20_000, // Poll every 2 seconds if the first sync is in progress, otherwise every 20 seconds
+    pollInterval: !importData?.lastSyncAt || importData.isSyncing || hasRequestedSync ? 2_000 : 20_000, // Poll every 2 seconds if syncing, otherwise every 20 seconds
     fetchPolicy: 'no-cache', // We want to always fetch the latest data, and make sure we don't update the cache
     notifyOnNetworkStatusChange: true, // To make sure `onCompleted` is called when the query is polled
     skip: !importData || hasNewData || !importData.connectedAccount || hasSettingsModal, // We can stop polling if we already know there's new data
     onCompleted(pollData) {
       if (!pollData.transactionsImport) {
         setIsDeleted(true);
         return;
-      } else if (importData.lastSyncAt !== pollData.transactionsImport.lastSyncAt) {
+      }
+
+      // Update the `isSyncing` status
+      if (importData.isSyncing !== pollData.transactionsImport.isSyncing) {
+        const newValue = pollData.transactionsImport.isSyncing;
+        apolloClient.cache.modify({
+          id: apolloClient.cache.identify(importData),
+          fields: { isSyncing: () => newValue },
+        });
+      }
+
+      // If we've manually requested a sync and a new one is registered
+      if (hasRequestedSync && importData.lastSyncAt !== pollData.transactionsImport.lastSyncAt) {
+        setHasRequestedSync(false);
+        apolloClient.cache.modify({
+          id: apolloClient.cache.identify(importData),
+          fields: { lastSyncAt: () => pollData.transactionsImport.lastSyncAt },
+        });
+      }
+
+      // Handle new sync data
+      if (importData.lastSyncCursor !== pollData.transactionsImport.lastSyncCursor) {
```

Comment:
This real-time pooling schema ended up getting quite complex.
In retrospect, perhaps it was simpler to have isolated all this logic within the sync button and let the cache do its work.

Or maybe we can create an abstract hook for real-time side effects using query pooling.
```ts
useQueryEffect(query, callback, {variables, context, pollInterval, skip});
```

Reply:
The sync button is not so clever, because the `syncPlaidAccount` mutation itself doesn't do much: it triggers a request on Plaid, but the real synchronization happens when Plaid calls back our Webhook (and it might actually never call it if there's nothing to sync).

The polling itself has the huge benefit of not relying on what triggered the sync: it could be the user, another user, a planned synchronization or a webhook event - in all cases, the page will react appropriately. I tend to disagree that the polling is complex: the reconciliation function (`onComplete`) is just 20 lines that handle 3 conditions:
- Dispatch `isSyncing` if it updates
- Dispatch `lastSyncAt` if it updates
- Add a banner with a refresh button if there's new data available

We could extract it in a dedicated hook to isolate the complexity somewhere else, but at this stage I wouldn't be confident implementing something as generic as a `useQueryEffect`.